lib/api/controllers/authController.js

```diff
@@ -231,21 +220,20 @@ class AuthController extends BaseController {
    * @param {Request} request
    * @returns {Promise<object>}
    */
-  checkToken(request) {
-    assertHasBody(request);
-    assertBodyHasAttribute(request, 'token');
+  checkToken (request) {
+    const token = this.getStringParam(request, 'body.token');
 
-    return this.kuzzle.repositories.token.verifyToken(request.input.body.token)
-      .then(token => Bluebird.resolve({
+    return this.kuzzle.repositories.token.verifyToken(token)
+      .then(token => ({
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') "token" hides or potentially hides a variable declared in an outer scope at line 224. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AVariableShadowing)